packages/vite/src/node/optimizer/index.ts

```diff
@@ -1199,13 +1198,23 @@ const lockfileFormats = [
   { name: 'pnpm-lock.yaml', checkPatches: false }, // Included in lockfile
   { name: 'bun.lockb', checkPatches: true },
 ]
+const lockfileNames = lockfileFormats.map((l) => l.name)
+
+function findNearestLockfile(dir: string) {
+  while (dir) {
+    for (const fileName of lockfileNames) {
+      const fullPath = path.join(dir, fileName)
+      if (tryStatSync(fullPath)?.isFile()) return fullPath
+    }
+    const parentDir = path.dirname(dir)
+    if (parentDir === dir) return
+
+    dir = parentDir
+  }
+}
```

Comment:
I think we can have a specialized version of the lookup here, this was the reason the second argument to `lookupFile` was an array in all other places.

Reply:
I think It'd still prefer the array version instead so we don't have two similar code 🤔 There shouldn't be a huge perf hit for the case where an array isn't needed too.

Reply:
done, the function isn't used much now anyways